In [13]:
import sys
sys.path.append('../src')  # So you can import from the src folder

from preprocessing import preprocess_reviews
import pandas as pd

# Define paths to your raw review CSV files
dashen_path = "../Data/Dashen_Bank_reviews_20250606_130804.csv"
cbe_path = "../Data/Commercial_Bank_of_Ethiopia_reviews_20250606_130816.csv"
boa_path = "../Data/Bank_of_Abyssinia_reviews_20250606_130818.csv"

# Define a utility function for comparison
def compare_cleaning(input_path, label):
    print(f"\n📊 Cleaning summary for {label}:")
    
    # Load original data
    original_df = pd.read_csv(input_path)
    original_shape = original_df.shape

    # Preprocess
    cleaned_df = preprocess_reviews(input_path)
    cleaned_shape = cleaned_df.shape if cleaned_df is not None else (0, 0)

    # Print comparison
    removed = original_shape[0] - cleaned_shape[0]
    print(f"Original shape: {original_shape}")
    print(f"Cleaned shape : {cleaned_shape}")
    print(f"❌ Rows removed: {removed}")

# Run cleaning and comparisons
compare_cleaning(dashen_path, "Dashen Bank")
compare_cleaning(cbe_path, "Commercial Bank of Ethiopia (CBE)")
compare_cleaning(boa_path, "Bank of Abyssinia (BOA)")



📊 Cleaning summary for Dashen Bank:
✅ Cleaned data saved to: ../Data\Dashen_Bank_reviews_20250606_130804_cleaned.csv
Original shape: (448, 5)
Cleaned shape : (447, 5)
❌ Rows removed: 1

📊 Cleaning summary for Commercial Bank of Ethiopia (CBE):
✅ Cleaned data saved to: ../Data\Commercial_Bank_of_Ethiopia_reviews_20250606_130816_cleaned.csv
Original shape: (5000, 5)
Cleaned shape : (4916, 5)
❌ Rows removed: 84

📊 Cleaning summary for Bank of Abyssinia (BOA):
✅ Cleaned data saved to: ../Data\Bank_of_Abyssinia_reviews_20250606_130818_cleaned.csv
Original shape: (1045, 5)
Cleaned shape : (1040, 5)
❌ Rows removed: 5


✅ Task 1: Review Data Preprocessing Summary
As part of the data quality requirements, I ensured:

Minimum 400 reviews per bank

Less than 5% missing or invalid data

📊 Cleaning Results:
Bank                 	                 Original_Reviews	    Cleaned_Reviews 	Rows Removed	  % Removed
Dashen Bank           	                  448	                     447	           1	             0.22%
Commercial Bank of Ethiopia (CBE)	      5000	                    4916         	  84	              1.68%
Bank of Abyssinia (BOA)	                  1045	                    1040         	   5	              0.48%

✅ Summary:
All three banks meet the minimum review count requirement (≥ 400).

All cleaned datasets have less than 5% removed entries, satisfying the <5% error tolerance requirement.

Cleaned data has been saved to CSV files and is now ready for sentiment and theme analysis.